projects/igniteui-angular/src/lib/grids/hierarchical-grid/child-grid-row.component.ts

```diff
@@ -169,7 +169,11 @@ export class IgxChildGridRowComponent implements AfterViewInit, OnInit {
         }
         const layouts = this.hGrid.childLayoutList.toArray();
         layouts.forEach((l) => this.hGrid.hgridAPI.registerLayout(l));
-        this.parentGrid.hgridAPI.registerChildGrid(this.rowData.rowID, this.layout.key, this.hGrid);
+        this.parentGrid.hgridAPI.registerChildGrid(this.rowData.rowID, this.layout.id, this.hGrid);
+        if (this.parentGrid.parent) {
```

Comment:
@skrustev Instead of registering everything on the root grid service and branching the logic that accessed the child grids it may be better to inject separate services for the layout instances and have each track its own children (currently layouts use the root grid's API service).